### DISCUSSION TITLE
Allow use of vipsblob from vips_profile_load () in vips_icc_import () & vips_icc_export () etc.

### DISCUSSION THREAD

**Discussion by sjards** [libvips/libvips#2407] [Ideas]
Hello,

I've been using vips_icc_import () & vips_icc_export () and they're working great for me, thank you!
I've looked at vips_profile_load () but can't see how to use the vipsblob it returns; sorry if I'm missing something (I'm limited to the exposed c api functions only).

Thought it would be good to add the ability to pass a loaded profile to the icc functions instead of always using file path/name.
Must admit, I'm assuming there would be a performance gain if processing may images, may be wrong with that.

Love the project!! many thanks

**Comment by jcupitt** [comment]
Hi @sjards, you can just attach the blob to the image as the icc profile and the `vips_icc_` functions will pick it up.

Though I don't think you'll see a performance improvement -- libvips has a profile cache and will reuse pointers behind your back.

**Comment by sjards** [comment]
Thanks @jcupitt! Missed that! Seems obvious now you've pointed it out :)
